syft/pkg/license.go

```diff
@@ -123,12 +128,14 @@ func NewLicenseFromFields(value, url string, location *file.Location) License {
 		l.Locations.Add(*location)
 	}
 	if url != "" {
-		l.URLs.Add(url)
+		l.URLs = append(l.URLs, url)
 	}
+
 	return l
 }
 
-// this is a bit of a hack to not infinitely recurse when hashing a license
+// Merge two licenses into a new license object.
+// TODO: this is a bit of a hack to not infinitely recurse when hashing a license
```

Comment:
I don't understand this comment. What's hacky about this, and how does it relate to recursing on hashing a package object? Do we merge licenses somewhere?

Reply:
I did not add this comment, but since I saw "hack" I wanted to encapsulate this with a `TODO`.

I think what this is commenting on is this exists to help with terminating the recursion of the `Hash` function used to calculate IDs, but am not entirely sure.

Reply:
This might benefit from a more specific doc comment on the function; I was surprised that "Merge" errors if the licenses aren't the same type, and only merges the location and URL fields.

Reply:
updated!